src/Propel/Generator/Model/Column.php

```diff
@@ -319,12 +319,12 @@ protected function setupObject()
                 $this->parentTable->getAttribute('columnPrefix'),
             );
 
-            // Accessor visibility
-            $visibility = $this->getMethodVisibility('accessorVisibility', 'defaultAccessorVisibility');
+            // Accessor visibility - no idea why this returns null, or the use case for that
+            $visibility = $this->getMethodVisibility('accessorVisibility', 'defaultAccessorVisibility') ?: '';
```

Comment:
But is continuing with an empty string better then?
Shouldnt this throw an exception or default to a sensible value instead?

Reply:
I'm all for throwing Exceptions.  But, this lib has a ton of business logic around null values.  I don't know if it's the tests, or these values are encountered during normal model generation, or what.  In many cases, null is being passed to a core string function that converts it to an empty string.  However, with PHP 8, that's not allowed, as there are more strict type checks.  In this case, we're just passing an empty string into `strtolower` instead of `null`.  But there are quite a few cases like this through the codebase.

The best solution here is to type our internal methods/properties.  This will prevent most of these issues and start to narrow down the actual source of the issue.

Reply:
True
But then I would rather have string|null here, what do you think?
As checking against null vs non empty string sounds a bit cleaner from the API level.

Reply:
I'm not really sure I'm following you.  `$visibility` ends up `null` in some cases.  It's then passed to `strtolower` which results in a type error on PHP 8.  Before PHP8, that'd just get converted to an empty string.

Reply:
Fair enough